src/github.com/getlantern/flashlight/flashlight.go

```diff
@@ -261,7 +260,14 @@ func runServerProxy(cfg *config.Config) {
 		}
 	}()
 
-	err = srv.ListenAndServe()
+	err = srv.ListenAndServe(func(update func(*server.ServerConfig) error) {
+		err := config.Update(func(cfg *config.Config) error {
+			return update(cfg.Server)
+		})
+		if err != nil {
+			log.Errorf("Error while trying to update: %v", err)
+		}
+	})
```

Comment:
ListenAndServe does block.

ListenAndServe only registers after successfully calling Listen(), which is really the thing that could fail.  Serve simply dispatches incoming requests.  There's no race condition because as soon as Listen was successful, the server will accept connections.  There might just be a slight delay before it reads from the connection.